server/src/main/java/org/elasticsearch/index/mapper/GeoShapeQueryable.java

```diff
@@ -43,10 +29,17 @@ public interface GeoShapeQueryable {
 
     Query geoShapeQuery(SearchExecutionContext context, String fieldName, ShapeRelation relation, LatLonGeometry... luceneGeometries);
 
-    default Query geoShapeQuery(SearchExecutionContext context, String fieldName, ShapeRelation relation, Geometry shape) {
+    default Query geoShapeQuery(SearchExecutionContext context, String fieldName, ShapeRelation relation, Geometry geometry) {
+        final Consumer<ShapeType> checker = relation == ShapeRelation.WITHIN ? t -> {
+            if (t == ShapeType.LINESTRING) {
+                // Line geometries and WITHIN relation is not supported by Lucene. Throw an error here
+                // to have same behavior for runtime fields.
```

Comment:
Even if not supported by lucene, I presume in ES|QL that simply means we cannot optimize this case by pushing down to lucene, but could still support it using source values.

Reply:
We cannot support it because of quantization. We are changing the language not the indexing model.